bugsnag-android-core/src/test/java/com/bugsnag/android/EventFilenameTest.kt

```diff
@@ -49,6 +49,66 @@ internal class EventFilenameTest {
         }
     }
 
+    @Test
+    fun testFindLaunchCrashReportInvalid() {
+        val eventStore = EventStore(
+            BugsnagTestUtils.generateImmutableConfig(),
+            NoopLogger,
+            Notifier(),
+            BackgroundTaskService(),
+            FileStore.Delegate { _, _, _ -> }
+        )
+
+        // no files
+        assertNull(eventStore.findLaunchCrashReport(emptyList()))
+
+        // regular crash reports
+        val jvmCrashReport = File("1504255147933_683c6b92-b325-4987-80ad-77086509ca1e.json")
+        assertNull(eventStore.findLaunchCrashReport(listOf(jvmCrashReport)))
+        val ndkCrashReport =
+            File("1504255147933_0000111122223333aaaabbbbcccc9999_c_my-uuid-123_not-jvm.json")
+        assertNull(eventStore.findLaunchCrashReport(listOf(ndkCrashReport)))
+    }
+
+    @Test
+    fun testFindSingleLaunchCrashReport() {
+        val eventStore = EventStore(
+            BugsnagTestUtils.generateImmutableConfig(),
+            NoopLogger,
+            Notifier(),
+            BackgroundTaskService(),
+            FileStore.Delegate { _, _, _ -> }
+        )
+
+        // startup crashes
+        val expected = File("1504255147933_30b7e350-dcd1-4032-969e-98d30be62bbc_startupcrash.json")
+        assertEquals(expected, eventStore.findLaunchCrashReport(listOf(expected)))
+    }
+
+    @Test
+    fun testFindMultipleLaunchCrashReport() {
+        val eventStore = EventStore(
+            BugsnagTestUtils.generateImmutableConfig(),
+            NoopLogger,
+            Notifier(),
+            BackgroundTaskService(),
+            FileStore.Delegate { _, _, _ -> }
+        )
+
+        // if multiple crashes exist, pick the most recent one
+        val expected = File("1664219155431_042c6195-a32c-2f84-11ae-77086509ca1e_startupcrash.json")
+        assertEquals(
+            expected,
+            eventStore.findLaunchCrashReport(
+                listOf(
+                    File("1504255147933_30b7e350-dcd1-4032-969e-98d30be62bbc_startupcrash.json"),
+                    File("1404205127135_683c6b92-b325-4987-80ad-77086509ca1e_startupcrash.json"),
+                    expected
```

Comment:
Since the sorting and selection logic grabs an item from an end of the array, I'd put `expected` in the middle to ensure that it worked correctly. For example, if the following was mistakenly commented out, this test would still pass as-is:

https://github.com/bugsnag/bugsnag-android/blob/c20f08c3ee31db3f7aa77d25d828bcbbb62e9a6d/bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java#L125

```suggestion
                    File("1504255147933_30b7e350-dcd1-4032-969e-98d30be62bbc_startupcrash.json"),
                    expected,
                    File("1404205127135_683c6b92-b325-4987-80ad-77086509ca1e_startupcrash.json")
```